### PR TITLE
Modify generate sourcemaps script

### PR DESCRIPTION
**Pull request by mateioprea**
Modify generate sourcemaps script in order to work with typescript files.

### CODE REVIEW

**Comment by mateioprea** [comment]
@joshuap hi, I don't know what the contributions guidelines are for this project but can you help with reviewing this PR? 

For RN you can also use a index.tsx / index.ts file as main file, and running the current generate source maps script doesn't do anything if you have a ts file. 

Thanks! :)

File: honeybadger-generate-sourcemaps.sh [hunk 1]
```diff
@@ -13,11 +13,27 @@
...
 PROJECT_ROOT_DIR="$SCRIPT_DIR/../../"
+INDEX_FILE="$PROJECT_ROOT_DIR/index.js"
+
+# Check if index.js file exists and if try to search for .ts or .tsx equivalent
+
+if [[ ! -f "$INDEX_FILE" ]]; then
+	if [[ -f "$PROJECT_ROOT_DIR/index.tsx" ]]; then
+		INDEX_FILE="$PROJECT_ROOT_DIR/index.tsx"
+	else
+		if [[ -f "index.ts" ]]; then
```
Comment (andreybutov) [inline honeybadger-generate-sourcemaps.sh:24]: Should this be `if [[ -f "$PROJECT_ROOT_DIR/index.ts" ]]; then` ?

Comment (mateioprea) [reply honeybadger-generate-sourcemaps.sh:24]: Yes, you're right!

**Comment by andreybutov** [comment]
I was doing some research into this, and I'm not sure we want to include this. 

Let me explain, and maybe you can help me understand.

The official RN docs state that index.js should remain index.js, even if the project uses TypeScript, as changing the entry point to .ts/x will cause issues with production bundling.

From https://reactnative.dev/docs/typescript:

Screen Shot 2021-07-27 at 12 37 36 PM

I dug deeper into this, and came across these two recent PRs suggesting that index.js be converted to index.ts for TypeScript-only projects. The PR comments indicate that RN doesn't have support for anything but index.js for an entry point, and that renaming it would cause issues for bundling:

https://github.com/react-native-community/react-native-template-typescript/issues/121
https://github.com/facebook/react-native/pull/27491

I also looked into some RN TypeScript templates, and also found that while App.js has been renamed to App.tsx, index.js remains index.js.

Finally, since index.js is just a simple call to registerComponent to mount App, there really shouldn't be any cause to use JSX or TypeScript in there. 

The technicals of the PR are correct, but I'm not sure adding this (and having to support the extra use case) would be prudent here.

But, once again, maybe you can help me understand. :)

**Comment by mateioprea** [comment]
Hi @andreybutov, you're correct. In my case I've migrated all files to typescript. 
Although it's not recommended changing the index file into a typescript file, if you do, you can specify it in the "Bundle React Native code and images" phase, in Build Phases. Check screenshot below. 

Maybe my case is unique and in this case my PR is not required. 😄 

Screenshot 2021-07-27 at 20 06 46

**Comment by andreybutov** [comment]
OK. We're going to hold off on pulling this in for now. Thanks!

**Comment by mateioprea** [comment]
No worries! Thank youu! 😄

**Pull request closed without merging**